src/plugins/data_source_management/public/components/data_source_menu/create_data_source_menu.tsx

```diff
@@ -6,21 +6,30 @@
 import React from 'react';
 import { EuiHeaderLinks } from '@elastic/eui';
 import { IUiSettingsClient } from 'src/core/public';
+import { DataSourcePluginSetup } from 'src/plugins/data_source/public';
 import { DataSourceMenu } from './data_source_menu';
 import { DataSourceMenuProps } from './types';
 import { MountPointPortal } from '../../../../opensearch_dashboards_react/public';
 
-export function createDataSourceMenu<T>(uiSettings: IUiSettingsClient) {
+export function createDataSourceMenu<T>(
+  uiSettings: IUiSettingsClient,
+  dataSource: DataSourcePluginSetup
+) {
   return (props: DataSourceMenuProps<T>) => {
+    const { hideLocalCluster } = dataSource;
     if (props.setMenuMountPoint) {
       return (
         <MountPointPortal setMountPoint={props.setMenuMountPoint}>
           <EuiHeaderLinks data-test-subj="top-nav" gutterSize="xs">
-            <DataSourceMenu {...props} uiSettings={uiSettings} />
+            <DataSourceMenu
+              {...props}
+              uiSettings={uiSettings}
+              hideLocalCluster={hideLocalCluster}
+            />
           </EuiHeaderLinks>
         </MountPointPortal>
       );
     }
-    return <DataSourceMenu {...props} />;
+    return <DataSourceMenu {...props} hideLocalCluster={hideLocalCluster} />;
```

Comment:
Looks like we missed adding the uiSettings option here in https://github.com/opensearch-project/OpenSearch-Dashboards/commit/726fb0e34704dbd54c71f9e716654323d67a302e, could you help to add uiSettings here, to test, please use the example plugin for top nav menu https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/examples/multiple_data_source_examples/public/components/data_source_via_top_nav_menu.tsx

Reply:
Added. However when I tried to test it by using `DataSourceSelectable` for top nav menu, it doesn't show "default" label for any options, Here's the recording.

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/4756cfa1-6836-46cb-9542-619355e88462


I think the issue is with that in top_nav_menu.tsx,  `DataSourceMenu` is not obtained by calling the `dataSourceManagement.ui.getDataSourceMenu<DataSourceSelectableConfig>();`, but through a direct import as below code suggests. 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a0eaf8483cbc29daf95d559c4b98965ad168052d/src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx#L44

Therefore, even I added the `uiSetting` to `createDataSourceMenu()`, it makes no difference for top nav menu use case, because `ui.getDataSourceMenu` API is never called, and `uiSetting` is never being passed through.
@BionIT What do you think? I can file an issue if it makes sense

Reply:
Right, yeah, could you help to cut an issue for that?